rns_shell/platform/graphics/x11/WindowX11.cpp

```diff
@@ -53,18 +58,44 @@ void Window::createEventLoop(Application* app) {
     const int x11_fd = ConnectionNumber(display);
 
     bool done = false;
+    auto screenDimension = RnsShell::PlatformDisplay::sharedDisplay().screenSize();
+
+    RnsShell::PlatformDisplay::sharedDisplay().setCurrentScreenSize(screenDimension.width(),screenDimension.height());
+
     while (!done) {
         if (int count = XPending(display)) {
             while (count-- && !done) {
                 XEvent event;
                 XNextEvent(display, &event);
-
                 switch (event.type) {
-                    case ConfigureNotify:
-                        RNS_LOG_INFO("Resize Request with (Width x Height) : (" << event.xconfigurerequest.width <<
+                    case ConfigureNotify:{
+
+                        if(event.xconfigurerequest.window == DefaultRootWindow(display)) {
+                            RNS_LOG_INFO(" ROOT Window(Screen) Size :" << event.xconfigurerequest.width << "x" << event.xconfigurerequest.height);
+                            auto screenSize =RnsShell::PlatformDisplay::sharedDisplay().getCurrentScreenSize();
+
+                            if(((screenSize.width()!=event.xconfigurerequest.width) ||
+                                (screenSize.height()!=event.xconfigurerequest.height))) {
+
+                                RnsShell::PlatformDisplay::sharedDisplay().setCurrentScreenSize(event.xconfigurerequest.width,event.xconfigurerequest.height);
+                                NotificationCenter::defaultCenter().emit("dimensionEventNotification");
+
+                            }
+
+                        } else {
+                                RNS_LOG_INFO("Resize Request with (Width x Height) : (" << event.xconfigurerequest.width <<
                                     " x " << event.xconfigurerequest.height << ")");
-                        app->sizeChanged(event.xconfigurerequest.width, event.xconfigurerequest.height);
+
+                                if((Window::currentWindowDimension.width()!=event.xconfigurerequest.width) ||
+                                    (Window::currentWindowDimension.height()!=event.xconfigurerequest.height)) {
```

Comment:
You should not use static method and data here. We can add root window to the hash map like other window and use following code to retrieve the window object.

WindowX11* win = WindowX11::gWindowMap.find(event.xany.window)